near/kycdao-ntnft/src/lib.rs

```diff
@@ -152,21 +253,35 @@ impl KycdaoNTNFT {
         // Get prefilled metadata, also remove digest so it cannot be used again
         let metadata = self.authorized_token_metadata.remove(&digest).expect("Unauthorized code");
         let status = self.authorized_statuses.remove(&digest).unwrap_or_default();
+        let tier = self.authorized_tiers.remove(&digest).unwrap_or("".to_string());
+        let seconds_to_pay = self.authorized_seconds_to_pay.remove(&digest).unwrap_or(0);
+
+        let cost = self.get_required_mint_cost_for_seconds_internal(seconds_to_pay);
+        if cost > 0 {
+            assert!(near_sdk::env::attached_deposit() >= cost, "Insufficient payment for minting");
+        }
 
         let token_id = self.next_token_id;
         self.next_token_id = self.next_token_id.checked_add(1).expect("Token ID overflow");
         let token_id_str = token_id.to_string();
 
-        let token = self.tokens.internal_mint(token_id_str.clone(), dst, Some(metadata), None);
+        let token = self.tokens.internal_mint(token_id_str.clone(), dst, Some(metadata), Some(cost));
```

Comment:
Was just reading the NTNFT code, looks like the refund (amount sent - cost) is handled there, just confirming?

Reply:
Yes, it should handle the refund, that's why we pass the cost here, so it could calculate the refund amount